src/Pkg.jl

```diff
@@ -231,17 +231,16 @@ const update = API.up
 !!! compat "Julia 1.9"
     `allow_reresolve` requires at least Julia 1.9.
 
-Run the tests for package `pkg`, or for the current project (which thus needs to be a package) if no
-positional argument is given to `Pkg.test`. A package is tested by running its
-`test/runtests.jl` file.
+Run the tests for package `pkg`, or  if no positional argument is given to `Pkg.test` for the current project
+(which thus needs to be a package). A package is tested by running its `test/runtests.jl` file.
 
-The tests are run by generating a temporary environment with only the `pkg` package
-and its (recursive) dependencies in it. If a manifest file exists and the `allow_reresolve`
-keyword argument is set to `false`, the versions in the manifest file are used.
-Otherwise a feasible set of packages is resolved and installed.
+The tests are run in a temporary environment that also includes the test specific dependencies
+of the package. The versions of dependencies in the current project are used for the
+test environment unless there is a compatibility conflict between the version of the dependencies and
+the test-specific dependencies. In that case, if `allow_reresolve` is `false` an error is thrown and
+if `allow_reresolve` is `true` a feasible set of versions of the dependencies is resolved and used.
```

Comment:
It'd be good to add a note about yanked versions here. It's caught me out before when I've expected a manifest to be observed.
```suggestion
The tests are run in a temporary environment that also includes the test specific dependencies
of the package. The versions of dependencies in the current project are used for the
test environment unless there is a compatibility conflict between the version of the dependencies and
the test-specific dependencies (or if the version in the manifest has been "yanked" from the registry and 
is thus marked as unavailable). In that case, if `allow_reresolve` is `false` an error is thrown and if 
`allow_reresolve` is `true` a feasible set of versions of the dependencies is resolved and used.
```

Reply:
bump @KristofferC the change here is `(or if the version in the manifest has been "yanked" from the registry and 
is thus marked as unavailable)`